proto/sedachain/core/v1/genesis.proto

```diff
@@ -10,11 +10,19 @@ option go_package = "github.com/sedaprotocol/seda-chain/x/core/types";
 // GenesisState defines core module's genesis state.
 message GenesisState {
   Params params = 1 [ (gogoproto.nullable) = false ];
-  bool paused = 2;
-  string owner = 3 [ (cosmos_proto.scalar) = "cosmos.AddressString" ];
-  string pending_owner = 4 [ (cosmos_proto.scalar) = "cosmos.AddressString" ];
-  uint64 committing_count = 5;
-  uint64 revealing_count = 6;
-  uint64 tallying_count = 7;
-  uint32 staker_count = 8;
+  string owner = 2 [ (cosmos_proto.scalar) = "cosmos.AddressString" ];
+  string pending_owner = 3 [ (cosmos_proto.scalar) = "cosmos.AddressString" ];
+  bool paused = 4;
+
+  // Staking-related states:
+  repeated string allowlist = 5
+      [ (cosmos_proto.scalar) = "cosmos.AddressString" ];
+  repeated Staker stakers = 6 [ (gogoproto.nullable) = false ];
+
+  // Data request-related states:
+  repeated DataRequest data_requests = 7 [ (gogoproto.nullable) = false ];
+  repeated RevealBody reveal_bodies = 8 [ (gogoproto.nullable) = false ];
```

Comment:
Oh so we have to store these in the genesis file? Isn't that going to be expensive since that's read/writing to a JSON file? How come a store isn't enough for them?

Reply:
Or is it just initialization purposes? Sorry I don't know much about how the genesis section really works yet :sweat_smile: